edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/RegistryController.cs

```diff
@@ -0,0 +1,735 @@
+        readonly Task<IEdgeHub> edgeHubGetter;
+
+        public RegistryController(
+            IRegistryApiClient apiClient,
+            Task<IEdgeHub> edgeHubGetter,
+            Task<IHttpRequestAuthenticator> authenticatorGetter)
+        {
+            this.apiClient = Preconditions.CheckNotNull(apiClient, nameof(apiClient));
+            this.authenticatorGetter = Preconditions.CheckNotNull(authenticatorGetter, nameof(authenticatorGetter));
+            this.edgeHubGetter = Preconditions.CheckNotNull(edgeHubGetter, nameof(edgeHubGetter));
+        }
+
+        [HttpPut]
+        [Route("devices/{deviceId}/modules/{moduleId}")]
+        public async Task CreateOrUpdateModuleAsync(
+            [FromRoute] string deviceId,
+            [FromRoute] string moduleId,
+            [FromBody] Module module)
+        {
+            try
+            {
+                Events.ReceivedRequest(deviceId, moduleId);
+
+                try
+                {
+                    deviceId = WebUtility.UrlDecode(Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId)));
+                    moduleId = WebUtility.UrlDecode(Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId)));
+                    Preconditions.CheckNotNull(module, nameof(module));
+
+                    if (!string.Equals(deviceId, module.DeviceId) || !string.Equals(moduleId, module.Id))
+                    {
+                        throw new ApplicationException("Device Id or module Id doesn't match between request URI and body.");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Events.BadRequest(ex.Message);
+                    await this.SendResponseAsync(HttpStatusCode.BadRequest, this.FormatErrorResponseMessage(ex.Message));
+                    return;
+                }
+
+                IHttpRequestAuthenticator authenticator = await this.authenticatorGetter;
+                HttpAuthResult authResult = await authenticator.AuthenticateAsync(deviceId, Option.None<string>(), this.HttpContext);
+
+                if (!authResult.Authenticated)
+                {
+                    Events.AuthenticateFail(deviceId);
+                    await this.SendResponseAsync(HttpStatusCode.Unauthorized);
+                    return;
+                }
+
+                Events.Authenticated(deviceId);
+                IEdgeHub edgeHub = await this.edgeHubGetter;
+                string edgeDeviceId = edgeHub.GetEdgeDeviceId();
+                IDeviceScopeIdentitiesCache identitiesCache = edgeHub.GetDeviceScopeIdentitiesCache();
+
+                (bool authorized, string authChain) = await this.AuthorizeActorAsync(identitiesCache, edgeDeviceId, Constants.EdgeHubModuleId, deviceId);
+                if (!authorized)
+                {
+                    await this.SendResponseAsync(HttpStatusCode.Unauthorized);
+                    return;
+                }
```

Comment:
this is probably unnecessary, AuthZ is only needed a child Edge acting OnBehalfOf a grandchild.  For the legacy endpoint, the current-Edge is always authorized to act OnBehalfOf of any child that connects to us.